sktime/transformations/series/detrend/_deseasonalize.py

```diff
@@ -517,23 +523,49 @@ def transform(self, Z, X=None):
         """
         self.check_is_fitted()
         z = check_series(Z, enforce_univariate=True)
-        return self._transform(z)
+        return self._inverse_transform(z)
 
-    def inverse_transform(self, Z, X=None):
-        """Inverse transform data.
+    def _make_return_object(self, X, stl):
 
-        Returns a transformed version of y.
+        # deseasonalize only
+        transformed = pd.Series(X.values - stl.seasonal, index=X.index)
```

Comment:
as discussed in my PR #1677 we wanted to make this flexible with another parameter, so user could chose between `method="deseasonalize"` and `method="detrend"`

Reply:
hm, but aren't the components all returned? The user can then choose by choosing a column.

Reply:
this is not working when you wnat to put the transformer in a pipeline